src/averageAges.js

```diff
@@ -55,7 +68,29 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const childrenArray = onlyWithSon
+    ? people.filter((x) =>
+      (people.some(haveMother => x.mother === haveMother.name)
+      && (x.sex === 'm')))
+    : people.filter((x) =>
+      (people.some(haveMother => x.mother === haveMother.name)));
+
+  const womenNames = childrenArray.map(child => child.mother);
+  const womenArray = [];
+
+  womenNames.forEach((x, index) => {
+    people.filter((item) => {
+      if (x === item.name) {
```

Comment:
Avoid using nesting and `if` statement
![image](https://github.com/mate-academy/js_average-ages/assets/70009205/8bfc7878-9b30-40e9-9444-88bf5362773a)